### PR TITLE
chore(docs): add an Error handling sample

### LINKED ISSUE #83: Improve the Error handling sample and add more documentation
https://github.com/googleapis/nodejs-datastore/blob/master/samples/error.js

It would be good to have actual example of error handling, and/or complete documentation about what errors can occur and what they mean.

### PR DESCRIPTION
**Pull request by vishald123**
Fixes #83  (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by vishald123** [comment]
Pull request for the following issue: 
https://github.com/googleapis/nodejs-datastore/issues/83

**Comment by stephenplusplus** [comment]
Looks good. So, did we delete this sample file at some point, and now we're bringing it back?

**Comment by AVaksman** [comment]
> Looks good. So, did we delete this sample file at some point, and now we're bringing it back?

It was deleted some time ago #110.
And now it looks like there is a request to add it back ([link](https://github.com/googleapis/nodejs-datastore/issues/83#issuecomment-386156741))

**Comment by stephenplusplus** [comment]
How can we make sure this new sample will have a home on the docs site and not be axed again?

**Comment by AVaksman** [comment]
Perhaps bring it up on the chat and ask Frank Natividad?

**Comment by stephenplusplus** [comment]
cc @frankyn!

**Comment by frankyn** [comment]
Summoning my colleague @benwhitehead, who is my point person for Datastore/Firestore related DevEx.

**Comment by BenWhitehead** [comment]
I'll poke around in docsite and figure out how/where this was published before and circle back.

**Comment by AVaksman** [comment]
@googlebot I consent.

File: samples/error.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const {Datastore} = require('@google-cloud/datastore');
+
+// [START error]
+function runQuery() {
+  // Creates a client
+  const datastore = new Datastore();
+
+  const query = datastore.createQuery(['Company']).start('badrequest');
+
+  return datastore
```
Comment (bcoe) [inline samples/error.js]: we tend to use an `async` function called `main` these days, here's a sample from the guidelines document:

```js
function main(arg1 = 'default_value_one', arg2 = 1234) {
  // [START your_region_tag]
  /**
   * TODO(developer): Uncomment these variables before running the sample.
   */
  // const arg1 = 'default_value_one';
  // const arg2 = 1234;

  // Imports the Google Cloud Some API library
  const {SomeApiClient} = require('@google-cloud/some-api');

  // Instantiates a client
  const someApiClient = new SomeApiClient();

  async function doSomething() {
    ...
  }

  doSomething();
  // [END your_region_tag]
}

main(...process.argv.slice(2));
```

**Pull request merged**